src/Analyzer/QueryTreeBuilder.cpp

```diff
@@ -233,6 +233,17 @@ QueryTreeNodePtr QueryTreeBuilder::buildSelectExpression(const ASTPtr & select_q
     auto select_settings = select_query_typed.settings();
     SettingsChanges settings_changes;
 
+    if (is_subquery)
```

Comment:
This is not right. Because `buildSelectExpression` can be called for subquery but without is_subquery, for example for `view(SELECT 1)`, `SELECT 1` is not subquery.
I suspect that we need to add local workaround in this place, and do not handle it anywhere else. This `limit`, `offset` settings does not look easily understandable when we send them to remote servers.
Here we need to save `limit`, `offset` settings, and remove them from our our updated context.
Then we can get minimum between this `limit` and LIMIT in query if it exists using `if` function. If LIMIT does not exists we can just add constant node with `limit` setting instead of LIMIT.
For `offset`, we must use `plus` function with OFFSET in query if it exists. If OFFSET does not exists we can just add constant node with `offset` setting instead of OFFSET.

Reply:
`view` argument receives its context with `getViewContext` function - I fixed it there - I added test for `view` and it works